src/chat/chatter.py

```diff
@@ -482,3 +487,15 @@ def joinInGame(self, url):
 
     def viewReplay(self, url):
         replay(url)
+
+    def view_in_leaderboards(self):
```

Comment:
I can't accept all this code being put in the chatter class, what you're doing here is responsibility of the tournament tab, not the chatter. Put tournament tab code in the tournament tab and expose an interface to it, refactor code if necessary to do so. Same goes for client and accessing / switching tabs, details of it should be in the client class and exposed to the outside as functions, maybe we can expose tabs as member of some enum.

Once there, make sure we don't use magic numbers or URLs, put them in constants or in settings.

Reply:
tournament ?

Reply:
Leaderboard tab, my mistake.

Reply:
I don't understand what you want,
 could you explain it on how it is done on viewVaultReplay (some lines above)?

Reply:
The code in viewVaultReplay is also really ugly, but there's no reason to perpetuate this ugliness. Point is (taking viewVaultReplay as example), the chatter should not need to know about private details like replays.setCurrentIndex or replays.vaultManager. This is old code that violates principle of single responsibility.

What we should do is put methods that do leaderboards stuff in the whatever class 'leaderboards' is, and call it with whatever it needs (in this case, the user). Same for the tab setting in client, there should be a method that sets the tab (maybe based on some enum) rather than directly manipulating its mainTabs member (and while we're at it, some nicer way of grabbing tabs too, maybe also based on some enum).

Reply:
well, I tried some ...

Reply:
what about: 'self.chat_widget.client....' visa 'client.instance....'  ?
(sheo once recommented  to use the later throughout)

Reply:
It doesn't matter, the thing that matters is that we don't fiddle with other objects' (de facto) private members, and instead move any parts of this code that do into these objects as methods.

Reply:
so in practice chatter should call a func in _clientwindow which calls a function in _replaywidget?

Reply:
It's going in the right direction, I'd add a dedicated method in _clientwindow for setting active tab (maybe using some enum instead of numbers), and a dedicated 'show this user in leaderboards' method in leaderboards widget (cramming it into busy_entered is not a good idea, since busy_entered is meant to be called only when we switch to the tab). Both should be called from the chatter - we can grab the tournament widget directly from client as attribute, or add a method in client for accessing different tabs if it would feel prettier.

Reply:
guess that should do it.